### DISCUSSION TITLE
Confusion with ESP32-S3

### DISCUSSION THREAD

**Discussion by Huckies** [hathach/tinyusb#1323] [Q&A]
Hello everyone. I managed to rewrite the CMakeLists of example hid_composite_freertos and it works, now I want to create two interfaces instead of one, but it keeps failing. Win10 throws out a problem status 0xC000000E. I'm a little confused with the usb things, a 'composite device' should refer to a single device with two interface descriptors instead of two HID descriptors, shouldn't it?

**Comment by Huckies** [comment]
@me-no-dev Sorry for the inconvenience I can only publish here. I'm more confused because according to the documentation [https://www.espressif.com/sites/default/files/documentation/esp32-s3_datasheet_cn.pdf](url) up to five IN EDs can be concurrent, excluding the ED0 I can still have four. So if I config two more EDs as OUT, can't these six Endpoints(excluding ED0) operate at the same time?

**Reply by me-no-dev** [reply]
From that TRM:

• Maximum of five IN endpoints concurrently active at any time (including EP0 IN)
• All OUT endpoints share a single RX FIFO
• Each IN endpoint has a dedicated TX FIFO

So excluding EP0, you have 4 more IN endpoints that you can assign FIFOs to. You can use all 7 OUT endpoints as they have their own shared FIFO.

**Reply by Huckies** [reply]
What about mixing? Like combination of 4*IN and 2*OUT? Will it work?
